hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java

```diff
@@ -805,51 +805,79 @@ public boolean isBucketEmpty(String volume, String bucket)
     return deletedKeys;
   }
 
+  /**
+   * @param userName volume owner, null for listing all volumes.
+   * @throws IOException
+   */
   @Override
   public List<OmVolumeArgs> listVolumes(String userName,
       String prefix, String startKey, int maxKeys) throws IOException {
     List<OmVolumeArgs> result = Lists.newArrayList();
-    UserVolumeInfo volumes;
+    List<UserVolumeInfo> volumes = Lists.newArrayList();
```

Comment:
This list is unnecessary.  `getVolumesByUser` returns a single `UserVolumeInfo`, so the list can only have exactly 1 item.

Reply:
Thanks @adoroszlai for the review.
I leave the original code for this part.
Fixed